synedrion/src/uint/traits.rs

```diff
@@ -16,152 +18,70 @@ pub trait ToMontgomery: Integer {
     }
 }
 
-/// Exponentiation functions for generic integers (in our case used for integers in Montgomery form
-/// with `Signed` exponents).
-pub trait Exponentiable<T>:
-    PowBoundedExp<T> + Invert<Output = CtOption<Self>> + ConditionallySelectable + Square + core::ops::Mul<Output = Self>
+pub trait Exponentiable<Exponent> {
```

Comment:
Why delete the docs here and in the rest of the file? At least the panicking behaviour and CT-nature should be documented imo.

Reply:
Probably lost as I was gradually replacing the old trait, my bad

Reply:
Added back the trait docs.